EntranceShuffle.py

```diff
@@ -1262,77 +1997,133 @@ def simple_shuffle_dungeons(world, player):
     dungeon_entrances = ['Eastern Palace', 'Tower of Hera', 'Thieves Town', 'Skull Woods Final Section', 'Palace of Darkness', 'Ice Palace', 'Misery Mire', 'Swamp Palace']
     dungeon_exits = ['Eastern Palace Exit', 'Tower of Hera Exit', 'Thieves Town Exit', 'Skull Woods Final Section Exit', 'Palace of Darkness Exit', 'Ice Palace Exit', 'Misery Mire Exit', 'Swamp Palace Exit']
 
-    if not world.shuffle_ganon:
-        connect_two_way(world, 'Ganons Tower', 'Ganons Tower Exit', player)
+    if world.mode != 'inverted':
+        if not world.shuffle_ganon:
+            connect_two_way(world, 'Ganons Tower', 'Ganons Tower Exit', player)
+        else:
+            dungeon_entrances.append('Ganons Tower')
+            dungeon_exits.append('Ganons Tower Exit')
     else:
-        dungeon_entrances.append('Ganons Tower')
-        dungeon_exits.append('Ganons Tower Exit')
+        dungeon_entrances.append('Inverted Agahnims Tower')
+        dungeon_exits.append('Inverted Agahnims Tower Exit')
 
     # shuffle up single entrance dungeons
     connect_random(world, dungeon_entrances, dungeon_exits, player, True)
 
     # mix up 4 door dungeons
     multi_dungeons = ['Desert', 'Turtle Rock']
-    if world.mode == 'open':
+    if world.mode in ['open', 'inverted'] and world.shuffle_ganon is True:
```

Comment:
hmm... Should this acually be 
`mode == 'open' or (mode == 'inverted' and shuffle_ganon)`?

Reply:
Yeah, that makes sense

Reply:
I also did something similar with an extra unecessary branch about 5 lines down that led to the same line of code